ietf/community/tests.py

```diff
@@ -108,80 +119,85 @@ def test_view_list(self):
             state=State.objects.get(type="draft", slug="active"),
             text="test",
         )
-        r = self.client.get(url)
-        self.assertEqual(r.status_code, 200)
-        self.assertContains(r, draft.name)
+        for id in self.email_or_name_set(person):
+            url = urlreverse(ietf.community.views.view_list, kwargs={ "email_or_name": id })
+            r = self.client.get(url)
+            self.assertEqual(r.status_code, 200)
+            self.assertContains(r, draft.name)
 
     def test_manage_personal_list(self):
-
-        person = PersonFactory(user__username='plain')
+        person = self.complex_person(user__username='plain')
         ad = Person.objects.get(user__username='ad')
         draft = WgDraftFactory(authors=[ad])
 
         url = urlreverse(ietf.community.views.manage_list, kwargs={ "email_or_name": person.email() })
         login_testing_unauthorized(self, "plain", url)
 
-        page = self.app.get(url, user='plain')
-        self.assertEqual(page.status_int, 200)
-
-        # add document
-        self.assertIn('add_document', page.forms)
-        form = page.forms['add_document']
-        form['documents'].options=[(draft.pk, True, draft.name)]
-        page = form.submit('action',value='add_documents')
-        self.assertEqual(page.status_int, 302)
-        clist = CommunityList.objects.get(person__user__username="plain")
-        self.assertTrue(clist.added_docs.filter(pk=draft.pk))
-        page = page.follow()
-
-        self.assertContains(page, draft.name)
-
-        # remove document
-        self.assertIn('remove_document_%s' % draft.pk, page.forms)
-        form = page.forms['remove_document_%s' % draft.pk]
-        page = form.submit('action',value='remove_document')
-        self.assertEqual(page.status_int, 302)
-        clist = CommunityList.objects.get(person__user__username="plain")
-        self.assertTrue(not clist.added_docs.filter(pk=draft.pk))
-        page = page.follow()
+        for id in self.email_or_name_set(person):
+            url = urlreverse(ietf.community.views.manage_list, kwargs={ "email_or_name": id })
+            #debug.show('id')
```

Comment:
Yes, it's quite nice.

As a warning about an issue that has bitten me more than once: if you use Python's native `assert` statement, it also takes an optional message, but if you try to use it like `assert(some_condition, "Some condition should be true")`, then it will *always* pass. That one needs to be `assert some_condition, "Some condition should be true"` because `assert` is a statement and does not use parentheses. If you put them in, you're asserting a tuple!